sdk/storage/azdatalake/filesystem/client.go

```diff
@@ -202,7 +203,15 @@ func (fs *Client) BlobURL() string {
 // The new directory.Client uses the same request policy pipeline as the Client.
 func (fs *Client) NewDirectoryClient(directoryPath string) *directory.Client {
 	directoryPath = strings.ReplaceAll(directoryPath, "\\", "/")
-	dirURL := runtime.JoinPaths(fs.generatedFSClientWithDFS().Endpoint(), directoryPath)
+	names := strings.Split(directoryPath, "/")
```

Comment:
As this code is being duplicated lets move this to another function and call that common method from all instances where we need to escape.

Reply:
moved changes to shared.go